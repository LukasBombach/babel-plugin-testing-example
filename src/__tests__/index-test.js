const babel = require('babel-core');
const plugin = require('../');
const assert = require('assert');
const printAst = require('ast-pretty-print');

const example = `
const foo = 1;
if (foo) console.log(foo);
`;

it('works', () => {
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

it('contains baz', () => {
  const {ast} = babel.transform(example, {plugins: [plugin]});
  const program = ast.program;
  const declaration = program.body[0].declarations[0];
  assert.equal(declaration.id.name, 'baz');
  expect(printAst(ast)).toMatchSnapshot();
});

