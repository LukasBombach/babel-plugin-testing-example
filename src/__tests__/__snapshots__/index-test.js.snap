exports[`test works 1`] = `
"
const baz = 1;
if (baz) console.log(baz);"
`;
